clients/iceberg-extensions/src/main/scala/org/apache/spark/sql/execution/datasources/v2/ShowLogExec.scala

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (C) 2020 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.sql.execution.datasources.v2
+
+import org.apache.spark.sql.catalyst.InternalRow
+import org.apache.spark.sql.catalyst.expressions.Attribute
+import org.apache.spark.sql.catalyst.util.{ArrayBasedMapData, MapData}
+import org.apache.spark.sql.connector.catalog.CatalogPlugin
+import org.apache.spark.unsafe.types.UTF8String
+import org.projectnessie.api.params.CommitLogParams
+import org.projectnessie.client.{NessieClient, StreamingUtil}
+
+import java.time.Instant
+import java.time.temporal.ChronoUnit
+import scala.collection.JavaConverters._
+
+case class ShowLogExec(
+    output: Seq[Attribute],
+    branch: Option[String],
+    currentCatalog: CatalogPlugin,
+    catalog: Option[String]
+) extends V2CommandExec {
+
+  lazy val nessieClient: NessieClient =
+    NessieUtils.nessieClient(currentCatalog, catalog)
+
+  override protected def run(): Seq[InternalRow] = {
+    val refName = branch.getOrElse(
+      NessieUtils.getCurrentRef(currentCatalog, catalog).getName
+    )
+    val stream = StreamingUtil.getCommitLogStream(
+      nessieClient.getTreeApi,
+      refName,
+      CommitLogParams.empty()
+    )
+
+    stream.iterator.asScala
+      .map(
+        cm =>
+          InternalRow(
+            convert(cm.getAuthor),
```

Comment:
This is interesting. I wonder whether we could have something like `SHOW committer, commit_time FROM LOG` or even fancier `SHOW committer, commit_time, substr(message, 1, 50) FROM LOG`.  (Not yet, but in general).
Or whether it's maybe better to make the log some "virtual table" that you can use in a SELECT?

Reply:
The log returns a table. So we can do `SELECT commiter, commit_time, substr(message, 1, 50) FROM (SHOW LOG) where commiter="rymurr"`